lib/schema.js

```diff
@@ -579,6 +580,23 @@ Schema.prototype.add = function add(obj, prefix) {
   return this;
 };
 
+Schema.prototype.removeIndex = function removeIndex(index) {
+  if (arguments.length > 1 || Object.keys(arguments[0]).length > 1 && typeof arguments[0] !== 'string') {
```

Comment:
Not sure why there's `Object.keys(arguments[0]).length > 1` here, you should be able to call `removeIndex({ name: 1, age: 1 })` to remove a compound index on `{ name, age }`